### DISCUSSION TITLE
Just enabling new architecture for fabric component results in Undefined symbol: _RTNMyPdfDisplayCls

### DISCUSSION THREAD

**Discussion by BraveEvidence** [reactwg/react-native-new-architecture#118] [Q&A] [answered]
I created a new react native project. I have created fabric components for the new architecture previously as well but suddenly from yesterday I am facing issue saying `Undefined symbol: _RTNMyPdfDisplayCls` when I run `yarn iOS`. This issue is only for iOS. I have not written any objectivec code yet but just created the interface file in typescript but I have added the package.json file and pod spec file and ran the below commands

Also I have tried cleaning my iOS project multiple times, try deleting node_modules,yarn.lock, pods and pod file.lock as well

`bundle install && RCT_NEW_ARCH_ENABLED=1 bundle exec pod install`

`yarn add ./RTNMyPdfDisplay`

```
node myappiospdfurl/node_modules/react-native/scripts/generate-codegen-artifacts.js \
  --path myappiospdfurl/ \
  --outputPath myappiospdfurl/RTNMyPdfDisplay/generated/
```

`RCT_NEW_ARCH_ENABLED=1 bundle exec pod install `

Now when I run yarn iOS I get error

Full project over [here](https://github.com/BraveEvidence/FabricIssue) 

The app should at least run even though I have not written any objective c code but did the setup for my fabric component

Let me know what I am missing. [Here](https://www.youtube.com/watch?v=Rq-mAjkG2ME&list=PLQhQEGkwKZUrempLnmxjt7ZCZJu1W3p2i&index=2) is a proof of fabric components working for me previously
Check screenshot for below error message
<img width="857" alt="Screenshot 2023-03-07 at 8 27 08 PM" src="https://user-images.githubusercontent.com/24708929/223459658-6b54e5ad-c5d2-4257-8539-8f98fecc3b9b.png">

<img width="877" alt="Screenshot 2023-03-07 at 8 27 18 PM" src="https://user-images.githubusercontent.com/24708929/223459718-1e7046f1-e438-4426-9610-c7fce3e613e5.png">

**Comment by cipolleschi** [comment, marked as answer]
Hi @BraveEvidence, and thank you for your issue.
Your project can't work until you actually write some Objective-C.

When installing the pods, with `RCT_NEW_ARCH_ENABLED=1 bundle exec pod install` you are actually running Codegen. You can see it in the logs, where you should see something like this:
```
[Codegen] Adding script_phases to React-Codegen.
[Codegen] Generating ./build/generated/ios/React-Codegen.podspec.json
[Codegen] Processing react-native core libraries
[Codegen] Found react-native


[Codegen] >>>>> Searching for codegen-enabled libraries in /Users/cipolleschi/Official/react-native/packages/rn-tester/node_modules


[Codegen] >>>>> Searching for codegen-enabled libraries in react-native.config.js


[Codegen] >>>>> Searching for codegen-enabled libraries in the app
[Codegen] Found @react-native/tester


[Codegen] >>>>> Processing FBReactNativeSpec
[Codegen] Generated schema: /var/folders/7c/qm29xjcn3kd75by5ty28skpm0000gn/T/FBReactNativeSpecG4mNYG/schema.json
[Codegen] Generated artifacts: /Users/cipolleschi/Official/react-native/packages/rn-tester/build/generated/ios


[Codegen] >>>>> Processing rncore
[Codegen] Generated schema: /var/folders/7c/qm29xjcn3kd75by5ty28skpm0000gn/T/rncoreUzqNRJ/schema.json
[Codegen] Generated artifacts: /Users/cipolleschi/Official/react-native/packages/rn-tester/build/generated/ios


[Codegen] >>>>> Processing AppSpecs
[Codegen] Generated schema: /var/folders/7c/qm29xjcn3kd75by5ty28skpm0000gn/T/AppSpecsjmySXj/schema.json
[Codegen] Generated artifacts: /Users/cipolleschi/Official/react-native/packages/rn-tester/build/generated/ios


>>>>> Creating component provider
Generated schema list: /var/folders/7c/qm29xjcn3kd75by5ty28skpm0000gn/T/rn-tmp-schema-list.json
Generated provider in: /Users/cipolleschi/Official/react-native/React/Fabric
```

Among the various `>>>>> Processing XXX`, you should see your library, so a line with `>>>>> Processing rtn-my-pdf-display`.

Codegen actually generates some Objective-C and C++ code for you and it expect for you to fill in the blanks of what it requires, among which there is that method that needs to be implemented.
Specifically, you can see a file called `RCTThirdPartyFabricComponentsProvider.h` and `RCTThirdPartyFabricComponentsProvider.mm` in the React-Codegen Development pod that try to use that function.
